ports/rp2/machine_i2c.c

```diff
@@ -34,6 +34,20 @@
 
 #define DEFAULT_I2C_FREQ (400000)
 
+#ifndef PICO_DEFAULT_I2C
```

Comment:
If/when the config files for this robot are merged to `pico-sdk`, will they require this option to be set?  It could be that `pico-sdk` doesn't build without these defaults defined.

Because of that, I think it would be better to have a specific config option here that forces scl/sda to be provided by the user.  Eg `MICROPY_HW_I2C_NO_DEFAULT_PINS`.

Reply:
It's already merged.  I think that definition is just used in one place in pico-sdk, to define some other things:

https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_i2c/include/hardware/i2c.h#L71-L77

But sure, we can add the extra macro and maybe it will prevent problems where future board file authors just forget to define their defaults.